internal/topology/check/compatibility.go

```diff
@@ -176,10 +177,153 @@ func LocalObjectTemplateIsValid(in *clusterv1.LocalObjectTemplate, namespace str
 		allErrs = append(allErrs,
 			field.Invalid(
 				pathPrefix.Child("ref", "apiVersion"),
-				in.Ref.APIVersion,
+				template.Ref.APIVersion,
 				"value cannot be empty",
 			),
 		)
 	}
 	return allErrs
 }
+
+// ClusterClassesAreCompatible checks the compatibility between new and old versions of a Cluster Class.
+// It checks that:
+// 1) Infrastructure Templates are compatible.
```

Comment:
Nit: I assume it's referring to InfrastructureClusterTemplates? (and not InfrastructureMachineTemplates)